src/backend/main/py.main/caliopen_main/objects/base.py

```diff
@@ -0,0 +1,447 @@
+            return main_errors.PatchUnprocessable(message="unable to unmarshall"
+                                                          " patch into object")
+        try:
+            self.get_db()
+        except NotFound as exc:
+            return exc
+
+        # check if patch is consistent with db current state
+        # if it is, squash self attributes
+        self.unmarshall_db()
+        for key in patch.keys():
+            if type(self._attrs[key]) is ListType:
+                if getattr(obj_patch_old, key) == [] and \
+                                getattr(self, key) != []:
+                    return main_errors.PatchConflict(message=
+                                                     "Patch current_state not consistent with db")
+                if getattr(self, key) == [] and \
+                                getattr(obj_patch_old, key) != []:
+                    return main_errors.PatchConflict(message=
+                                                     "Patch current_state not consistent with db")
+                for old in getattr(obj_patch_old, key):
+                    found = False
+                    for elem in getattr(self, key):
+                        if issubclass(self._attrs[key][0], CaliopenObject):
+                            if elem.__dict__ == old.__dict__:
+                                found = True
+                                break
+                        else:
+                            if elem == old:
+                                found = True
+                                break
+                    if not found:
+                        return main_errors.PatchConflict(message=
+                                                         "Patch current_state not consistent with db")
+            elif type(self._attrs[key]) is DictType:
+                if cmp(getattr(obj_patch_old, key), getattr(self, key)) != 0:
+                    return main_errors.PatchConflict(message=
+                                                     "Patch current_state not consistent with db")
+            else:
+                if getattr(obj_patch_old, key) != getattr(self, key):
+                    return main_errors.PatchConflict(message=
+                                                     "Patch current_state not consistent with db")
+
+            setattr(self, key, getattr(obj_patch_new, key))
+
+        if "db" in options and options["db"] is True:
+            log.info("will update db")
+            # apply changes to db model and update db
+            self.marshall_db()
+            try:
+                self.update_db()
+            except Exception as exc:
+                log.info(exc)
+                return main_errors.PatchError(message="Error when updating db")
+
+        if "index" in options and options["index"] is True:
+            # silently update index. Should we raise an error if it fails ?
+            try:
+                self.update_index()
```

Comment:
you can't call a method that will be define in a children class due to how ObjectUser <- ObjectIndexable inherit.

Define an apply_patch method on ObjectIndexable, call apply_patch of parent ObjectUser and then do specific processing as it's an indexable object